### PR TITLE
Support MultiQC reports in ChIP-seq workflows

### PR DESCRIPTION
**Pull request by jkokosar**
## Checklist

* [ ] Update CHANGELOG.rst for each commit separately:
  * Pay attention to write entries under the "Unreleased" section.
  * Mark all breaking changes as "**BACKWARD INCOMPATIBLE:**" and put them
    before non-breaking changes.
  * If a commit modifies a feature listed under "Unreleased" section,
    it might be sufficient to modify the existing CHANGELOG entry from previous
    commit(s).
* [ ] Bump the process version:
  * **MAJOR version (first number)**: Backward incompatible changes (changes
    that break the api/interface). Examples: renaming the input/output, adding
    mandatory input, removing input/output...
  * **MINOR version (middle number)**: add functionality or changes in a
    backwards-compatible manner. Examples: add output field, add non-mandatory
    input parameter, use a different tool that produces same results...
  * **PATCH version (last number)**: changes/bug fixes that do not affect
    the api/interface. Examples: typo fix, change/add warning messages...
* [ ] All inputs are used in process.
* [ ] All output fields have their ``re-save`` calls.

### Processes ###

* Code that was once accepted should be kept like it is. It should be changed
  only when its functionality has changed. Small parts may be changed
  to fix bugs (by changing code only to make it nicer you might introduce a bug).
* Set the optimal number of cores that the tool can use (check if parallelization
  can be performed).
* Description of tool should be self explanatory with links to tool's homepage
  and publication.
* Write explanatory input labels and add short command line parameter names
  in square brackets (for example: Maximum length [``--maxlength``]).
* Where possible provide default values for the process input fields.
  If they are provided, then ``required: false`` is not needed.
* Add descriptive error and warning messages.
* Place a ``re-checkrc`` call after each process step (call of a tool, a
  Bash command, a Python script or an R script).
* Add ``re-progress`` calls where appropriate.

### Commit and PR messages ###

* Each commit should be minimal (i.e. 1 change) and self-contained (including
  tests).
* Changes to Docker images need to be accepted in a separate pull request.
  Before the new images can be used for automated testing of your PR, they
  must be manually tagged and pushed to Docker Hub.
* Mark incomplete PRs with the [WIP] (Work In Progress) tag.
* Do not paste links to private repositories from the public one.

### Tests ###

* Add meaningful names of test files with spaces:
  * Genomes: ``genome species.fa.gz``
  * Annotations: ``annotation species.gtf.gz``
  * Adapters: ``adapters-name/type-source``.
* Output and specific input files should include the name of the tested tool.
* Keep test files' size small. If there is no other option, add the large
  test file to the ``files/large`` directory so it will be tracked with Git LFS.
* Pay attention to edge cases: what will happen if any data is
  missing, if length of some list is 0, if string is given where one expects
  integer, if file is empty, ...
* When testing workflows, try to verify workflow success by checking one
  of the last steps. Pick the one that is most likely to fail. Also test
  that all steps have 'OK' status.

### CODE REVIEW

File: resolwe_bio/processes/chip_seq/macs2_rose2_batch.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@
           - case
         required: True
   data_name: ChIP-Seq Workflow Batch
-  version: 1.0.3
+  version: 1.1.0
```
Comment (romunov) [inline resolwe_bio/processes/chip_seq/macs2_rose2_batch.yml]: For this file, only the order of option changes and `promoter` hidden status changes. Does this warrant a middle version bump?

File: resolwe_bio/processes/workflows/chip_seq.yml [hunk 2]
```diff
@@ -357,3 +357,341 @@
...
+        - name: q_threshold
+          label: Quality filtering threshold
+          type: basic:integer
+          default: 30
+        - name: n_sub
+          label: Number of reads to subsample
+          type: basic:integer
+          default: 15000000
+        - name: tn5
+          label: TN5 shifting
```
Comment (romunov) [inline resolwe_bio/processes/workflows/chip_seq.yml]: Make N small perhaps, something along the lines of `Tn5` (also used in description and on wikipedia)?

File: resolwe_bio/processes/workflows/chip_seq.yml [hunk 3]
```diff
@@ -357,3 +357,341 @@
...
+          choices:
+            - label: "1"
+              value: "1"
+            - label: "auto"
+              value: "auto"
+            - label: "all"
+              value: "all"
+          description: |
+            It controls the MACS behavior towards duplicate tags at the exact same location -- the
+            same coordination and the same strand. The 'auto' option makes MACS calculate the
```
Comment (romunov) [inline resolwe_bio/processes/workflows/chip_seq.yml]: coordination -> coordinate?

File: resolwe_bio/processes/workflows/chip_seq.yml [hunk 4]
```diff
@@ -357,3 +357,341 @@
...
+          required: false
+          default: all
+          choices:
+            - label: "1"
+              value: "1"
+            - label: "auto"
+              value: "auto"
+            - label: "all"
+              value: "all"
+          description: |
```
Comment (romunov) [inline resolwe_bio/processes/workflows/chip_seq.yml]: Same choices and description as for the `duplicate` argument. Is this correct?

File: resolwe_bio/processes/workflows/chip_seq.yml [hunk 5]
```diff
@@ -357,3 +357,341 @@
...
+            region is set too small, a sharp spike in the input data may kill the significant
+            peak.
+        - name: extsize
+          label: extsize
+          type: basic:integer
+          required: false
+          description: |
+            While '--nomodel' is set, MACS uses this parameter to extend reads in 5'->3' direction
+            to fix-sized fragments. For example, if the size of binding region for your
+            transcription factor is 200 bp, and you want to bypass the model building by MACS,
```
Comment (romunov) [inline resolwe_bio/processes/workflows/chip_seq.yml]: Sometimes bp is written without a space (i.e. 200bp) and with space (200 bp). Is this something you feel we need to standardize?

File: resolwe_bio/processes/workflows/chip_seq.yml [hunk 6]
```diff
@@ -357,3 +357,341 @@
...
+            this parameter can be set as 200. This option is only valid when --nomodel is set or
+            when MACS fails to build model and --fix-bimodal is on.
+        - name: shift
+          label: Shift
```
Comment (romunov) [inline resolwe_bio/processes/workflows/chip_seq.yml]: Perhaps a bit more informative label? "Shift in bp"?

File: resolwe_bio/processes/workflows/chip_seq.yml [hunk 7]
```diff
@@ -357,3 +357,341 @@
...
+          default: false
+          description: |
+            With this flag on, MACS will use the background lambda as local lambda. This means
+            MACS will not consider the local bias at peak candidate regions.
+        - name: fix_bimodal
+          label: Turn on the auto paired-peak model process
+          type: basic:boolean
+          default: false
+          description: |
+            Whether turn on the auto paired-peak model process. If it's set, when MACS failed
```
Comment (romunov) [inline resolwe_bio/processes/workflows/chip_seq.yml]: I think you can remove "Whether" -> "Turn on auto-peak model process.".

File: resolwe_bio/processes/workflows/chip_seq.yml [hunk 8]
```diff
@@ -357,3 +357,341 @@
...
+            to build paired model, it will use the nomodel settings, the '--extsize' parameter
+            to extend each tags. If set, MACS will be terminated if paired-peak model is failed.
```
Comment (romunov) [inline resolwe_bio/processes/workflows/chip_seq.yml]: "tags" -> "tag"
"is failed" -> "has failed"

File: resolwe_bio/processes/workflows/chip_seq.yml [hunk 9]
```diff
@@ -357,3 +357,341 @@
...
+        - name: nomodel
+          label: Bypass building the shifting model
+          type: basic:boolean
+          hidden: "tagalign"
+          default: false
+          description: |
+            While on, MACS will bypass building the shifting model.
+        - name: nomodel_prepeak
+          label: Bypass building the shifting model
```
Comment (romunov) [inline resolwe_bio/processes/workflows/chip_seq.yml]: Same label and description as for `nomodel`. Is this right?

File: resolwe_bio/processes/workflows/chip_seq.yml [hunk 10]
```diff
@@ -357,3 +357,341 @@
...
+          default: true
+          hidden: "!tagalign"
+          description: |
+            While on, MACS will bypass building the shifting model.
+        - name: down_sample
+          label: Down-sample
+          type: basic:boolean
+          default: false
+          description: |
+            When set, random sampling method will scale down the bigger sample. By default, MACS
```
Comment (romunov) [inline resolwe_bio/processes/workflows/chip_seq.yml]: "When set to true...".

File: resolwe_bio/processes/workflows/chip_seq.yml [hunk 11]
```diff
@@ -357,3 +357,341 @@
...
+            uses linear scaling. This option will make the results unstable and irreproducible
+            since each time, random reads would be selected, especially the numbers (pileup,
+            pvalue, qvalue) would change. Consider to use 'randsample' script before MACS2 runs
```
Comment (romunov) [inline resolwe_bio/processes/workflows/chip_seq.yml]: Is this script `randsample` even applicable in this workflow?

File: resolwe_bio/processes/workflows/chip_seq.yml [hunk 12]
```diff
@@ -357,3 +357,341 @@
...
+          description: |
+            Cutoff for broad region. This option is not available unless --broad is set. If -p is
+            set, this is a p-value cutoff, otherwise, it's a q-value cutoff. DEFAULT = 0.1
+  run:
+    language: workflow
+    program:
+      - id: macs2
+        run: macs2-callpeak
+        input:
+          case: '{{input.case}}'
```
Comment (romunov) [inline resolwe_bio/processes/workflows/chip_seq.yml]: In some other workflows, we use the syntax `{{ input.case }}` (space between contents and brackets).

File: resolwe_bio/processes/workflows/chip_seq.yml (hunk 4 above)
Comment (jkokosar) [reply resolwe_bio/processes/workflows/chip_seq.yml]: Yes, this is intentional. Depending on the choice of `prepeak` input, different UI elements are shown to the user, with different default values.

File: resolwe_bio/processes/workflows/chip_seq.yml (hunk 5 above)
Comment (jkokosar) [reply resolwe_bio/processes/workflows/chip_seq.yml]: Using a space is the correct way of writing units. Fixed in the related processes.

File: resolwe_bio/processes/workflows/chip_seq.yml (hunk 9 above)
Comment (jkokosar) [reply resolwe_bio/processes/workflows/chip_seq.yml]: Yes, the same situation as above.

File: resolwe_bio/processes/workflows/chip_seq.yml (hunk 3 above)
Comment (jkokosar) [reply resolwe_bio/processes/workflows/chip_seq.yml]: I have left this the same as is in use in the Macs2 manual.
